inc/file.class.php

```diff
@@ -276,8 +276,11 @@ public function post_addItem() {
    public function post_getFromDB() {
       // Check the user can view this itemtype and can view this item
       if ($this->canView() && $this->canViewItem()) {
-         if (isAPI()
-            && (isset($_SERVER['HTTP_ACCEPT']) && $_SERVER['HTTP_ACCEPT'] == 'application/octet-stream'
+         $filename = FLYVEMDM_FILE_PATH . '/' . $this->fields['source'];
+         $isFile = is_file($filename);
+         $this->fields['filesize'] = ($isFile) ? fileSize($filename) : 0;
+         $this->fields['mime_type'] = ($isFile) ? mime_content_type($filename) : '';
```

Comment:
I don't get the point to add extra keys in ```$this->fields``` when ```$_SERVER['HTTP_ACCEPT'] == 'application/octet-stream'``` because the code will send the content of the file, not any JSON. Moreover the methid sendFile() wil set the appropriate HTTP headers related to file's metadata. 

I thought you wanted to get the file size when the request asks for the data from the DB only.